compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -685,10 +686,15 @@ object TypeOps:
       }
     }
 
+    def isPatternTypeSymbol(sym: Symbol) = !sym.isClass && sym.is(Case)
+
     // replace uninstantiated type vars with WildcardType, check tests/patmat/3333.scala
```

Comment:
This sounds similar to what `Prototypes.wildApprox` does, could it be used instead of instUndetMap?